### DISCUSSION TITLE
Which technologies we'll use for streaming?

### DISCUSSION THREAD

**Discussion by rennanprysthon** [filipemelo2002/screenshr-app#12] [General]
I think we could first choose the architecture and technologies for streaming, and then we can select the frameworks and libraries. 

For example, are we going to use WebRTC or some other tool?

**Comment by rennanprysthon** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API

I had taken a look at that documentation, but I didn't delve too deep into it.

**Reply by filipemelo2002** [reply]
There are plenty examples on the web, but basically, we need to setup a signaling serve (or use Firebase for that), and also, use these APIs on the FE to stream their screen...

https://vegibit.com/nodejs-webrtc-signalling-server/#:~:text=Signaling%20servers%20are%20used%20to,js%20and%20Socket.io.

**Reply by filipemelo2002** [reply]
I also found this article which explains how WebRTC works: https://blog.logrocket.com/webrtc-signaling-websocket-node-js/

**Reply by filipemelo2002** [reply]
There's also this video which helped me a lot to understand how we are going to do it: https://youtu.be/WmR9IMUD_CY?si=DUr_p37Zg8brsp4q

**Comment by filipemelo2002** [comment]
Hey, I just finished reading [this blog](https://tsh.io/blog/how-to-write-video-chat-app-using-webrtc-and-nodejs/) and it seems I already have everything figured out how the connection/streaming will work. 
We will need the following components:

- A Client (in this case, this repository's project)
- A Signaling server


Basically, we will use the WebRTC APIs to create an offer (think of offers as when you make a phone call), which will be carried by WebSocket to our Signaling server (let's call this event "send-offer"). The signaling server will forward this "offer"  to everyone else who is also at the party, and they will accept it by creating another object called offer-answer, which will be sent back through WebSocket (let's call this event "answer-offer"). Once we receive this answer, we will be able to stream the screen directly through webRTC, so our backend doesn't need to stream the content manually, the webRTC will take care of that for us.

**Reply by rennanprysthon** [reply]
Which signaling server we're going to use?

Or we are going to create on from scratch?

**Reply by filipemelo2002** [reply]
I think we could create one of our own
